abd-clam/src/core/graph/_graph.rs

```diff
@@ -4,53 +4,64 @@ use std::collections::{HashMap, HashSet};
-type EdgeSet<'a, T, U, D> = HashSet<&'a Edge<'a, T, U, D>>;
-type AdjacencyMap<'a, T, U, D> = HashMap<&'a Cluster<T, U, D>, ClusterSet<'a, T, U, D>>;
-type FrontierSizes<'a, T, U, D> = HashMap<&'a Cluster<T, U, D>, Vec<usize>>;
+/// A set of clusters with references to clusters in a graph.
+pub type ClusterSet<'a, U> = HashSet<&'a Cluster<U>>;
+
+/// A set of edges with references to edges in a graph.
+pub type EdgeSet<'a, U> = HashSet<&'a Edge<'a, U>>;
+
+/// A map that represents the adjacency relationship between clusters.
+pub type AdjacencyMap<'a, U> = HashMap<&'a Cluster<U>, ClusterSet<'a, U>>;
+
+/// A map that associates clusters with lists of frontier sizes.
+pub type FrontierSizes<'a, U> = HashMap<&'a Cluster<U>, Vec<usize>>;
 
 /// Two `Cluster`s have an `Edge` between them if they have overlapping volumes.
 ///
 /// In CLAM, all `Edge`s are bi-directional.
 #[derive(Debug, Clone)]
-pub(crate) struct Edge<'a, T: Number, U: Number, D: Dataset<T, U>> {
-    left: &'a Cluster<T, U, D>,
-    right: &'a Cluster<T, U, D>,
-    #[allow(dead_code)]
+pub struct Edge<'a, U: Number> {
+    /// A reference to the first `Cluster` connected by this `Edge`.
+    left: &'a Cluster<U>,
+    /// A reference to the second `Cluster` connected by this `Edge`.
+    right: &'a Cluster<U>,
+    #[allow(unused_mut, unused_variables, dead_code)]
+    /// The distance between the two `Cluster`s connected by this `Edge`.
     distance: U,
 }
 
-impl<'a, T: Number, U: Number, D: Dataset<T, U>> PartialEq for Edge<'a, T, U, D> {
+impl<'a, U: Number> PartialEq for Edge<'a, U> {
     fn eq(&self, other: &Self) -> bool {
         (self.left == other.left) && (self.right == other.right)
     }
 }
 
 /// Two `Edge`s are equal if they connect the same two `Cluster`s.
-impl<'a, T: Number, U: Number, D: Dataset<T, U>> Eq for Edge<'a, T, U, D> {}
+impl<'a, U: Number> Eq for Edge<'a, U> {}
 
-impl<'a, T: Number, U: Number, D: Dataset<T, U>> std::fmt::Display for Edge<'a, T, U, D> {
+impl<'a, U: Number> std::fmt::Display for Edge<'a, U> {
     fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
         write!(f, "{:} -- {:}", self.left, self.right)
     }
 }
 
-impl<'a, T: Number, U: Number, D: Dataset<T, U>> Hash for Edge<'a, T, U, D> {
+impl<'a, U: Number> Hash for Edge<'a, U> {
     fn hash<H: Hasher>(&self, state: &mut H) {
-        format!("{self}").hash(state)
+        format!("{self}").hash(state);
     }
 }
 
-impl<'a, T: Number, U: Number, D: Dataset<T, U>> Edge<'a, T, U, D> {
+impl<'a, U: Number> Edge<'a, U> {
     /// Creates a new `Edge` from the given `Cluster`s and the distance between
     /// them.
     ///
     /// It is upon the user to verify that the two `Cluster`s are close enough
     /// to have an edge between them.
-    #[allow(dead_code)]
-    pub fn new(left: &'a Cluster<T, U, D>, right: &'a Cluster<T, U, D>, distance: U) -> Self {
+    #[allow(unused_mut, unused_variables, dead_code)]
```

Comment:
Remove this macro from all code. Make the `Graph` and `Edge` structs public in `lib.rs`

Reply:
Done. I did need to allow dead code for about a dozen fields because of clippy complaining

Reply:
Ok. In general, let's limit the use of these macros as much as possible.

Reply:
When I was first trying to get through pre-commit checks clippy wanted most of those macros used. I deleted all of them and then went back and added any macros in where clippy complained. This time it was only about a dozen.